crates/stellar-scaffold-macro/src/asset.rs

```diff
@@ -31,38 +37,68 @@ pub fn parse_asset(str: &str) -> Result<(xdr::Asset, String), xdr::Error> {
     ))
 }
 
-pub fn generate_asset_id(
-    asset: &str,
-    network: &Network,
-) -> Result<(stellar_strkey::Contract, String), xdr::Error> {
-    let (asset, code) = parse_asset(asset).unwrap();
-    let network_id = xdr::Hash(network.id());
-    let preimage = xdr::HashIdPreimage::ContractId(xdr::HashIdPreimageContractId {
-        network_id,
-        contract_id_preimage: xdr::ContractIdPreimage::Asset(asset.clone()),
-    });
-    let preimage_xdr = preimage.to_xdr(xdr::Limits::none())?;
-    Ok((
-        stellar_strkey::Contract(Sha256::digest(preimage_xdr).into()),
-        code,
-    ))
+pub fn get_serialized_asset(asset: &str) -> Result<(String, Vec<u8>), xdr::Error> {
+    let (asset, code) = parse_asset(asset)?;
+
+    let mut data = Vec::new();
+    let cursor = Cursor::new(&mut data);
+    let mut limit = xdr::Limited::new(cursor, xdr::Limits::none());
+    asset.write_xdr(&mut limit)?;
+
+    Ok((code, data))
 }
 
 /// Generate the code to read the `STELLAR_NETWORK` environment variable
 /// and call the `generate_asset_id` function
-pub fn parse_literal(lit_str: &syn::LitStr, network: &Network) -> TokenStream {
-    let (contract_id, code) = generate_asset_id(&lit_str.value(), network).unwrap();
-    // let contract_id = format_ident!("\"{contract_id}\"");
-    let contract_id = contract_id.to_string();
+pub fn parse_literal(lit_str: &syn::LitStr) -> TokenStream {
+    let (code, data) = get_serialized_asset(&lit_str.value()).unwrap();
     let mod_name = format_ident!("{code}");
+    let test_mod_name = format_ident!("test_{code}");
+
+    let size = data.len();
+
     quote! {
+        #[allow(non_upper_case_globals)]
+        pub(crate) mod #test_mod_name {
+            use super::*;
+
+            /// Create a Stellar Asset Client for the asset which provides an admin interface
+            pub fn stellar_asset_client<'a>(env: &soroban_sdk::Env, sac:&soroban_sdk::testutils::StellarAssetContract) -> soroban_sdk::token::StellarAssetClient<'a> {
+                soroban_sdk::token::StellarAssetClient::new(&env, &sac.address())
+            }
+            /// Create a Stellar Asset Client for the asset which provides an admin interface
+            pub fn token_client<'a>(env: &soroban_sdk::Env, sac:&soroban_sdk::testutils::StellarAssetContract) -> soroban_sdk::token::TokenClient<'a> {
+                soroban_sdk::token::TokenClient::new(&env, &sac.address())
+            }
+
+            /// Registers a new SAC contract (to use in unit tests only)
+            pub fn register(env: &soroban_sdk::Env, admin: &soroban_sdk::Address) -> soroban_sdk::testutils::StellarAssetContract {
+                let sac = env.register_stellar_asset_contract_v2(admin.clone());
+                let cl = stellar_asset_client(env, &sac);
+                env.mock_all_auths();
+                cl.mint(admin, &1_000_000_000_i128);
+                sac
+            }
+
+            pub fn to_min_unit(float: f64) -> i128 {
```

Comment:
Floats are not allowed by the soroban VM. So would be good to have a test with a compiled wasm too.